site/src/pages/RoadmapPage/ImportZot4PlanPopup.tsx

```diff
@@ -9,28 +9,51 @@ import trpc from '../../trpc.ts';
 import { expandAllPlanners, makeUniquePlanName } from '../../helpers/planner';
 import spawnToast from '../../helpers/toastify';
 import helpImage from '../../asset/zot4plan-import-help.png';
+import { useTransferredCredits } from '../../hooks/transferCredits';
+import { setUserAPExams } from '../../store/slices/transferCreditsSlice';
+import { useIsLoggedIn } from '../../hooks/isLoggedIn';
 
 interface ImportZot4PlanPopupProps {
   saveRoadmap: (planner?: RoadmapPlan[]) => Promise<void>;
 }
 const ImportZot4PlanPopup: FC<ImportZot4PlanPopupProps> = ({ saveRoadmap }) => {
   const dispatch = useAppDispatch();
   const { darkMode } = useContext(ThemeContext);
+  const isLoggedIn = useIsLoggedIn();
   const [showModal, setShowModal] = useState(false);
   const [scheduleName, setScheduleName] = useState('');
   const [studentYear, setStudentYear] = useState('1');
   const [busy, setBusy] = useState(false);
   const allPlanData = useAppSelector(selectAllPlans);
+  const apExams = useTransferredCredits().ap;
 
   const obtainImportedRoadmap = async (schedName: string, currYear: string) => {
     // Get the result
     try {
-      const result = await trpc.zot4PlanImportRouter.getScheduleFormatted.query({
+      const [savedRoadmap, z4pApExams] = await trpc.zot4PlanImport.getScheduleFormatted.query({
```

Comment:
To reflect changes from the previous comment
```suggestion
      const { savedRoadmap, apExams: z4pApExams } = await trpc.zot4PlanImport.getScheduleFormatted.query({
```